contracts/utilities/UtilityChecker.sol

```diff
@@ -0,0 +1,179 @@
+//                                         :@@@@@@@#.:#@@@%-
+//                                         +@@@@@@-.*@@@*:
+//                                         #@@@@#.=@@@+.
+//                                         @@@@+-%@%=
+//                                        :@@@#%@%=
+//                                        +@@@@%-
+//                                        :#%%=
+//
+
+/**
+ *     NOTICE
+ *
+ *     The T-REX software is licensed under a proprietary license or the GPL v.3.
+ *     If you choose to receive it under the GPL v.3 license, the following applies:
+ *     T-REX is a suite of smart contracts implementing the ERC-3643 standard and
+ *     developed by Tokeny to manage and transfer financial assets on EVM blockchains
+ *
+ *     Copyright (C) 2023, Tokeny sàrl.
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity 0.8.26;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import { IClaimIssuer, IIdentity } from "@onchain-id/solidity/contracts/interface/IClaimIssuer.sol";
+import { IIdentityRegistry, IClaimTopicsRegistry, ITrustedIssuersRegistry } from "../registry/interface/IIdentityRegistry.sol";
+import { IModularCompliance } from "../compliance/modular/IModularCompliance.sol";
+import { IModule } from "../compliance/modular/modules/IModule.sol";
+import { IToken } from "../token/IToken.sol";
+import { IUtilityChecker } from "./IUtilityChecker.sol";
+
+
+contract UtilityChecker is IUtilityChecker, OwnableUpgradeable, UUPSUpgradeable {
+
+    function initialize() external initializer {
+        __Ownable_init();
+    }
+
+    /// @inheritdoc IUtilityChecker
+    /// @dev This function is not gas optimized and should be called only OFF chain.
+    function testTransfer(address _token, address _from, address _to, uint256 _amount) 
+        external view override returns (bool) {
+        IToken token = IToken(_token);
+
+        if (token.paused()) return false;
+
+        (bool frozen, ) = testFreeze(_token, _from, _to, _amount);
+        if (frozen) return false;
+
+        ComplianceCheckDetails [] memory details = testTransferDetails(_token, _from, _to, _amount);
```

Comment:
you check compliance, pause and freeze but not the eligibility in this function, which means that right now a transfer that would block at the eligibility level will return true, while it should return false. You need to check the eligibility on the receiver address (just need to call `isVerified` on the IR of the token

Reply:
```suggestion
        IIdentityRegistry ir = token.identityRegistry();
        if(!ir.isVerified(_to)) return false; 
        ComplianceCheckDetails [] memory details = testTransferDetails(_token, _from, _to, _amount);
```